### ISSUE TITLE
[Wallet Stability] The wallet section takes an inappropriate amount of time to open on Windows

### ISSUE THREAD

**Issue by noeliaSD** [status-im/status-desktop#15273]
### Description
It is not possible to open the **wallet section** in an appropriate amount of time on _Windows_ on release version **2.29**. Also tested on master commit 0645ed47123f742faa2e110bc6daf1f2320dedd8 and same result.

A _Windows_ user shared the following videos:
https://monosnap.com/file/uZDazzQy5gIdwdSeNWMErJr27aleQ1

https://monosnap.com/file/Ws1lKYcS0jZNOveXjwIbViUbvGdhD6

The user has the following setup:
- 4 wallet accounts
- 20-30 collectibles
- 20-30 assets

### Goals for this task
1. Start an initial investigation on which can be the root cause of the issue:
    - Determine if it happens only on windows OS
    - Determine which are the main componentes involved
    - Analyse the main offenders on _nim_ and _status-go_ (callstack details when it happens)

2. Build a plan to mitigate the issue.

**Comment by alexjba** [comment]
@xAlisher Could you please share the logs you're getting when the app is stuck?

I've tried to reproduce and indeed opening wallet is a bit slow (3 sec on both windows and MacOs) but nothing close to what you're experiencing.

**Comment by xAlisher** [comment]
Sure!
[app_20240620_174413.log](https://github.com/user-attachments/files/15916684/app_20240620_174413.log)

**Comment by anastasiyaig** [comment]
@alexjba i logged this that might be a duplicate of the current task https://github.com/status-im/status-desktop/issues/15316
could that be UI issue? i thought it is some heavy call on backend

**Comment by anastasiyaig** [comment]
and it is not windows related for sure

**Comment by anastasiyaig** [comment]
@xAlisher is that what you see? I cant access monosnap links for some reason. maybe they banned me :D


https://github.com/status-im/status-desktop/assets/82375995/e20cafa2-a45d-47e1-9061-051a470780a6

**Comment by alexjba** [comment]
> @alexjba i logged this that might be a duplicate of the current task #15316 
Yes, it's a duplicate! Glad you've found it. Maybe it helps with debugging

> could that be UI issue? i thought it is some heavy call on backend
Not sure yet..The backend call mentioned in #15316 is async. Should not block at all.

**Comment by anastasiyaig** [comment]
i tried in 2.29 just now and app froze for 1:42
https://drive.google.com/file/d/15ND_H1Zh-UyQCb4HoZ8lZnkkO5w069RK/view?usp=sharing

**Comment by Valentina1133** [comment]
I also reproduced it on my Mac (M2 Max, Ventura 13.5, 32 Gb)

**Comment by alexjba** [comment]
@anastasiyaig @Valentina1133 I've prepared a build with some extra info. It's based on 2.29 branch. Could you please give it a shot?
This build will produce some extra logs/data so I'll need 3 files:
1. `/Applications/Status.app/Contents/MacOS/procBenchmark.csv`
2. Application logs from Status folder
3. geth logs from Status folder

https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-240625-064307-886ea9-pr15318-aarch64.dmg

**Comment by alexjba** [comment]
@xAlisher There's a potential fix here. It's a 2.29 build so you can use it instead of the RC if it proves to fix the hang
https://status-im-desktop-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-240625-071805-2c680f-pr15321-x86_64.exe

Built in this PR https://github.com/status-im/status-desktop/pull/15321

**Comment by alexjba** [comment]
https://github.com/status-im/status-desktop/pull/15321 is confirmed to solve the issues. But it's removing a feature.

Adding a follow-up task to re-design the URL validation for collectibles:
https://github.com/status-im/status-desktop/issues/15331

**Comment by alexjba** [comment]
Closed by https://github.com/status-im/status-desktop/pull/15321
